pkg/query-service/app/http_handler.go

```diff
@@ -2464,13 +2464,58 @@ func (aH *APIHandler) execClickHouseGraphQueries(ctx context.Context, queries ma
 		wg.Add(1)
 		go func(name, query string) {
 			defer wg.Done()
-			seriesList, err := aH.reader.GetTimeSeriesResultV3(ctx, query)
 
-			if err != nil {
-				ch <- channelResult{Err: fmt.Errorf("error in query-%s: %v", name, err), Name: name, Query: query}
+			if strings.HasSuffix(name, constants.FirstQuerySuffix) {
+				// Graph query with limit - first query skip
 				return
+			} else if strings.HasSuffix(name, constants.SecondQuerySuffix) {
+				// Graph query with limit - second query
+
+				// get the first query result
+				actualName := strings.ReplaceAll(name, constants.SecondQuerySuffix, "")
+				list, err := aH.reader.GetListResultV3(ctx, queries[actualName+constants.FirstQuerySuffix])
+				if err != nil {
+					ch <- channelResult{Err: fmt.Errorf("error in query-%s: %v", name, err), Name: actualName + constants.FirstQuerySuffix, Query: queries[actualName+constants.FirstQuerySuffix]}
+					return
+				}
+
+				// create the filterset from the first query result
+				/*
+					put | error
+					put | debug
+				*/
+				// filter : ( method = 'put' AND level = 'error' ) OR ( method = 'put' AND level = 'debug' )
+				filterString := ""
+				l := len(list)
+				for i, row := range list {
+					tFilter := ""
+					for name, val := range row.Data {
+						tFilter = tFilter + fmt.Sprintf("%s = '%s' AND ", name, *val.(*string))
```

Comment:
> `*val.(*string) `

The value can't always be a string? Please use `ClickHouseFormattedValue`. I would `AND` and then trim right below. Instead, I would prepare a list of conditionals and use `strings.Join` which is cleaner. And finally, join those intermediary independent condition statements with `OR`. I find it easier to follow as a reader.

Reply:
Thanks have updated it.